shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -1029,6 +1088,13 @@ - (void)engineCallbackOnPreEngineRestart {
   }
 }
 
+- (void)onVSync:(uintptr_t)baton {
+  @synchronized(_vsyncWaiters) {
+    FlutterVSyncWaiter* waiter = [_vsyncWaiters objectForKey:@(kFlutterImplicitViewId)];
```

Comment:
I'm assuming we'll need to generalise this for multi-views, in which case, you'll want to add a `TODO` here to wire up the proper view ID with an issue link. You may need to file a new subissue under https://github.com/flutter/flutter/issues/142845 if nothing existing fits.

Reply:
I added a TODO for this. Right now it seems a bit unclear how exactly vsync is going to work with multiview. We might not have a viewId available. 

At least for the multiview MVP we'll probably use some heuristic for this - i.e. if views are on different displays use waiter for view on primary display, or with display with higher refresh. We'll cross that bridge when we get there, buteEither way this should be a minor change.